### PR TITLE
Fix building issue with Meatpack defined only on the second serial port

### PR DESCRIPTION
**Pull request by X-Ryl669**
### Description

Fix building issue. Github fails to create the PR once, so I've lost my previous logorrhea. See code for description.

### Requirements

LPC1769

### Configurations

See #21010 

### Related Issues

See #21010

### CODE REVIEW

**Comment by Snargaloo** [comment]
Would you need a sanity check for people who were using multiserial and then comment out a lower ordinance serial port?  

For example:
//#define SERIAL_PORT 0
#define SERIAL_PORT_2 3

Only one serial port is defined, but now the code will enable multi serial.

File: Marlin/src/inc/Conditionals_post.h [hunk 1]
```diff
@@ -2907,6 +2907,16 @@
   #define HAS_ROTARY_ENCODER 1
 #endif
 
+#if NUM_SERIAL == 1 && defined(SERIAL_PORT_2)
+  // There's a bug here while platform IO is computing the BUILD_FEATURES for its scripts. It's including
+  // common-dependencies.h that (force) defines NUM_SERIAL to 1
+  // In turn, because NUM_SERIAL is set to 1, the code below undefine MEATPACK
+  // if it's defined from MEATPACK_ON_SERIAL_PORT_2 and it's not build anymore (it fails at link step)
```
Comment (rhapsodyv) [inline Marlin/src/inc/Conditionals_post.h]: So we should move the definition of NUM_SERIAL out of the hal. Is it possible?

**Comment by X-Ryl669** [comment]
I consider it's an error not to define SERIAL_PORT. Yes, we could have sanity check for all possible errors, but it'd take years to write and maintain. So, for now, only the most frequent errors are checked.

File: Marlin/src/inc/Conditionals_post.h (hunk 1 above)
Comment (rhapsodyv) [reply Marlin/src/inc/Conditionals_post.h]: just move 
```
#ifdef SERIAL_PORT_2
  #define NUM_SERIAL 2
#else
  #define NUM_SERIAL 1
#endif
```
to `Conditionals_post.h`
And remove `#define NUM_SERIAL 1` from `buildroot/share/PlatformIO/scripts/common-dependencies.h`

Comment (X-Ryl669) [reply Marlin/src/inc/Conditionals_post.h]: Some HAL define serial port by themselves (like LPC176x that's defining the LCD_SERIAL_PORT and MMU_SERIAL_PORT). I'm not sure if it's counted in the NUM_SERIAL macro.

We could also have a single `#define SERIAL_PORTS { -1, 1, 2, 3 }` (leading to `constexpr int SerialPorts[] = { -1, 1, 2, 3 }` and then just `#define NUM_SERIAL COUNT(SerialPorts)`. 
However, constructing the serial chain from such a list will be a template monster... Or not. Maybe it's worth investigating.  

It's clear that it'll probably be easier than what we have currently with all this "#ifdef SERIAL_IS_SOMETHING #else if something else" chain.

Comment (rhapsodyv) [reply Marlin/src/inc/Conditionals_post.h]: `NUM_SERIAL`, as I remember, is only related to the presence of SERIAL_1 and SERIAL_2 ... the others serial have different meaning and handling ...

**Comment by Snargaloo** [comment]
> 
> 
> I consider it's an error not to define SERIAL_PORT. Yes, we could have sanity check for all possible errors, but it'd take years to write and maintain. So, for now, only the most frequent errors are checked.

I agree that code is the wrong approach to such an issue.  Perhaps tweaking the documentation of Configuration.h to clarify serial ports need to be enabled sequentially.  If a board has 3 serial ports, and user wants to attach host to port 3, the other two need to be defined as well.  It is not obvious currently.

File: Marlin/src/inc/Conditionals_post.h (hunk 1 above)
Comment (rhapsodyv) [reply Marlin/src/inc/Conditionals_post.h]: @thinkyhead can we move the define of `NUM_SERIAL` to `Conditionals_post.h`?

Comment (X-Ryl669) [reply Marlin/src/inc/Conditionals_post.h]: I'm not available this WE for development. On monday, I'll change the way you said and check it also fix the issue.

**Comment by GhostlyCrowd** [comment]
I had a random freeze this morning after 12 hours, its relative to this issue and this one I believe. More info posted here 

https://github.com/MarlinFirmware/Marlin/issues/21119

**Comment by thinkyhead** [comment]
I believed at some point that the HALs would be "consulted" before deciding which serial ports were available and how many there would be. But at this point the serial ports are guaranteed to be chosen by the user in the first config without any overrides, and then later we are checking settings against the HAL's requirements in each HAL's `serial.h` header.

All of which is to say, I've moved the serial checks and conditionals, including `NUM_SERIAL` to the earliest possible points, so it will now be picked up correctly by the dependencies script.

**Pull request merged**